test/e2e/util/util.go

```diff
@@ -47,6 +47,65 @@ func GenerateNamespace(t *testing.T, ctx context.Context, client crclient.Client
 	return namespace
 }
 
+// CreateCluster creates a new namespace and a HostedCluster in that namespace using
+// the provided options, and returns the newly created resource and a teardown
+// function intended to be called via defer in test setup code.
+func CreateCluster(t *testing.T, ctx context.Context, client crclient.Client, opts cmdcluster.Options, artifactDir string) (*hyperv1.HostedCluster, func(context.Context)) {
+	g := NewWithT(t)
+	start := time.Now()
+
+	// Set up a namespace to contain the hostedcluster
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: SimpleNameGenerator.GenerateName("e2e-clusters-"),
+			Labels: map[string]string{
+				"hypershift-e2e-component": "hostedclusters-namespace",
+			},
+		},
+	}
+	err := client.Create(ctx, namespace)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to create namespace")
+
+	// Define the hostedcluster and adjust options to match it
+	hc := &hyperv1.HostedCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace.Name,
+			Name:      SimpleNameGenerator.GenerateName("example-"),
+		},
+	}
+	opts.Namespace = namespace.Name
+	opts.Name = hc.Name
+	opts.InfraID = hc.Name
+
+	// Define a standard teardown function
+	teardown := func(ctx context.Context) {
```

Comment:
Nit:
```suggestion
	teardown := func(ctx context.Context) {
	...
	t.Cleanup(teardown)
```

Reply:
I think the main advantage is that you do not need to pass around the cleanup funcs. The context is always `context.Background()` so that argument seems useless anyways?